### PR TITLE
Support bracket enclosed names in TZ POSIX rules

### PR DESCRIPTION
**Pull request by eerhardt**
The POSIX time zone rules at the end of a tzfile have been updated to allow digits, plus signs and minus signs in the time zone names by enclosing the name in `<` and `>` characters. TimeZoneInfo's parsing logic was written using an older version of the spec, so it didn't have support for this format.

Fix https://github.com/dotnet/corefx/issues/16962

@tarekgh

### CODE REVIEW

**Comment by tarekgh** [comment]
LGTM. thanks @eerhardt

**Comment by eerhardt** [comment]
test OSX x64 Checked Build and Test
test Windows_NT x64 Release Priority 1 Build and Test

File: src/mscorlib/src/System/TimeZoneInfo.Unix.cs [hunk 1]
```diff
@@ -1189,8 +1189,29 @@ private static bool TZif_ParseMDateRule(string dateRule, out int month, out int
...
+            bool isBracketEnclosed = index < posixFormat.Length && posixFormat[index] == '<';
+            if (isBracketEnclosed)
+            {
+                // move past the opening bracket
+                index++;
+                
+                string result = TZif_ParsePosixString(posixFormat, ref index, c => c == '>');
+                
+                // move past the closing bracket
+                index++;
```
Comment (stephentoub) [inline src/mscorlib/src/System/TimeZoneInfo.Unix.cs]: Do we need to verify that it is in fact a closing brace? Is it possible with malformed input we might get here and increment even if there isn't one?

Comment (eerhardt) [inline src/mscorlib/src/System/TimeZoneInfo.Unix.cs]: The only case where TZif_ParsePosixString returns is if the "breakCondition" func returns true or it gets to the end of the string.  If it hits the end of the string, incrementing index one more past the length won't cause problems. But just to be safe I can add the check that the current char is the closing brace before incrementing.

**Comment by tarekgh** [comment]
test OSX x64 Checked Build and Test please
test Ubuntu x64 Checked Build and Test please
test Windows_NT x64 Debug Build and Test please

**Comment by eerhardt** [comment]
test OSX10.12 x64 Checked Build and Test

**Pull request merged**